### ISSUE TITLE
Crash with Java 14 and HikariCP

### ISSUE THREAD

**Issue by nezda** [glowroot/glowroot#738]
I have a large application that builds and runs with Java 14 and uses [HikariCP](https://github.com/brettwooldridge/HikariCP) for db connection pooling (also uses [querydsl](https://github.com/querydsl/querydsl/tree/QUERYDSL_3_6_3) and the [postgres driver](https://github.com/pgjdbc/pgjdbc/tree/REL42.2.5)). As soon as it attempts to access the database, it falls over. Here is the stacktrace up to the calling code:

```
java.lang.VerifyError: Stack map does not match the one at exception handler 171
Exception Details:
  Location:
    com/sun/proxy/$Proxy51.prepareStatement(Ljava/lang/String;II)Ljava/sql/PreparedStatement; @171: iload
  Reason:
    Type 'java/lang/Throwable' (current frame, locals[1]) is not assignable to 'java/lang/String' (stack map, locals[1])
  Current Frame:
    bci: @125
    flags: { }
    locals: { 'com/sun/proxy/$Proxy51', 'java/lang/Throwable', integer, integer, integer, 'org/glowroot/agent/bytecode/api/ThreadContextThreadLocal$Holder', 'org/glowroot/agent/bytecode/api/ThreadContextPlus', integer, 'org/glowroot/agent/plugin/api/Timer', 'java/lang/String' }
    stack: { 'java/lang/Throwable' }
  Stackmap Frame:
    bci: @171
    flags: { }
    locals: { 'com/sun/proxy/$Proxy51', 'java/lang/String', integer, integer, integer, 'org/glowroot/agent/bytecode/api/ThreadContextThreadLocal$Holder', 'org/glowroot/agent/bytecode/api/ThreadContextPlus', integer, 'org/glowroot/agent/plugin/api/Timer', 'java/lang/String' }
    stack: { 'java/lang/Throwable' }
  Bytecode:
    0000000: 0336 0401 3a05 013a 0602 3607 013a 082b
    0000010: 3a09 1905 c700 08b8 0093 3a05 1905 b600
    0000020: 9b59 3a06 c600 1f19 06b9 009e 0100 5936
    0000030: 0712 9f9f 0010 1906 129f b900 a302 0004
    0000040: a700 0403 3604 1504 9900 0a19 06b8 00f1
    0000050: 3a08 2ab4 0058 2ab2 010a 06bd 0070 5903
    0000060: 2b53 5904 1cb8 00bb 5359 051d b800 bb53
    0000070: b900 6004 00c0 00f5 a700 0ebf 4cbb 0067
    0000080: 592b b700 6abf 1504 9900 0959 1909 b800
    0000090: f815 0499 0008 1908 b800 fc15 0702 9f00
    00000a0: 0c19 0615 07b9 00a3 0200 b015 0499 0008
    00000b0: 1908 b800 fc15 0702 9f00 0c19 0615 07b9
    00000c0: 00a3 0200 bf
  Exception Handler Table:
    bci [82, 123] => handler: 123
    bci [82, 123] => handler: 123
    bci [82, 123] => handler: 123
    bci [82, 123] => handler: 124
    bci [82, 134] => handler: 171
  Stackmap Table:
    full_frame(@28,{Object[#2],Object[#124],Integer,Integer,Integer,Object[#149],Object[#151],Integer,Object[#236],Object[#124]},{})
    same_frame(@67)
    same_locals_1_stack_item_frame(@68,Integer)
    same_frame(@82)
    same_locals_1_stack_item_frame(@123,Object[#84])
    same_locals_1_stack_item_frame(@124,Object[#84])
    same_locals_1_stack_item_frame(@134,Object[#245])
    same_locals_1_stack_item_frame(@145,Object[#245])
    same_locals_1_stack_item_frame(@155,Object[#245])
    same_locals_1_stack_item_frame(@170,Object[#245])
    same_locals_1_stack_item_frame(@171,Object[#84])
    same_locals_1_stack_item_frame(@181,Object[#84])
    same_locals_1_stack_item_frame(@196,Object[#84])

	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3215)
	at java.base/java.lang.Class.getConstructor0(Class.java:3420)
	at java.base/java.lang.Class.getConstructor(Class.java:2165)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.build(Proxy.java:660)
	at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$0(Proxy.java:424)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:422)
	at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1015)
	at com.zaxxer.hikari.pool.ProxyConnection$ClosedConnection.getClosedConnection(ProxyConnection.java:518)
	at com.zaxxer.hikari.pool.ProxyConnection$ClosedConnection.<clinit>(ProxyConnection.java:493)
	at com.zaxxer.hikari.pool.ProxyConnection.close(ProxyConnection.java:252)
```

I checked Glowroot, opened it with Intellij, and briefly attempted to try to add a test (thought maybe to agent > integration-tests > org.glowroot.agent.tests.JdbcDriverIT ?) but quickly got put off by the Java 6 and shading requirements that didn't seem to work right in the IDE. Before our recent conversion from Java 8 → Java 14 we got a lot of value from Glowroot and would love to resume using it. I can provide any other details required to help resolve this.

**Comment by trask** [comment]
I suspect that `asm` dependency needs to be updated in order to fix Java 14 support

**Comment by nezda** [comment]
I see there's a 

```
<dependency>
    <groupId>org.ow2.asm</groupId>
    <artifactId>asm</artifactId>
    <version>8.0.1</version>
</dependency>
```

I'm happy to try to test but may need some guidance unless I just test with my app.

**Comment by ionutursuleanu** [comment]
Hi @nezda ,

Try this build https://github.com/ionutursuleanu/glowroot/releases/tag/v0.13.17-snapshot.2020.07.15

**Comment by nezda** [comment]
Identical error @ionutursuleanu

**Comment by ionutursuleanu** [comment]
That build included the updated asm dependency.  If that didn't worked I don't know what could we try next

**Comment by nezda** [comment]
Can yall offer any guidance on how i can make an integration / unit test that exercises the agent with Java 14 code?

**Comment by ionutursuleanu** [comment]
Hi @nezda ,

I made [this](https://github.com/glowroot/glowroot/files/4933816/demo.zip) spring-boot app in order to reproduce the error, but nothing. It's running fine with openjdk 14.0.1 & glowroot configured to monitor the main method. Can you also try this or make your own sample app the reproduce the error?

> Can yall offer any guidance on how i can make an integration / unit test that exercises the agent with Java 14 code?

I'm not that familiar with maven to help you with this

```
/Library/Java/JavaVirtualMachines/adoptopenjdk-14.jdk/Contents/Home/bin/java -javaagent:/Users/ionutursuleanu/glowroot/glowroot.jar -XX:TieredStopAtLevel=1 -noverify -Dspring.output.ansi.enabled=always -Dcom.sun.management.jmxremote -Dspring.jmx.enabled=true -Dspring.liveBeansView.mbeanDomain -Dspring.application.admin.enabled=true -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=49899:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/ionutursuleanu/IdeaProjects/demo/build/classes/java/main:/Users/ionutursuleanu/IdeaProjects/demo/build/resources/main:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-starter-data-jpa/2.3.1.RELEASE/5531a4cbd506f13a0b4483ed73c2e75a03d8da46/spring-boot-starter-data-jpa-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-starter/2.3.1.RELEASE/e0d28696fea064578cb01da346232284f922eba4/spring-boot-starter-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-starter-aop/2.3.1.RELEASE/f42e23091d29bde8d1356e45cf13ad32dae51437/spring-boot-starter-aop-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-starter-jdbc/2.3.1.RELEASE/800b64e76588ea88761ad9108cde762204805d66/spring-boot-starter-jdbc-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.data/spring-data-jpa/2.3.1.RELEASE/1b5e106add569913f0c776aca62f85d8e9ca8cee/spring-data-jpa-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/jakarta.transaction/jakarta.transaction-api/1.3.3/c4179d48720a1e87202115fbed6089bdc4195405/jakarta.transaction-api-1.3.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/jakarta.persistence/jakarta.persistence-api/2.2.3/8f6ea5daedc614f07a3654a455660145286f024e/jakarta.persistence-api-2.2.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.hibernate/hibernate-core/5.4.17.Final/3b90ecf6fe93a27a27de9671c9fb25d03ba3def7/hibernate-core-5.4.17.Final.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-aspects/5.2.7.RELEASE/eb48f4ae3e1525179e1ccd10c0e09cfe5c27b8bb/spring-aspects-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-autoconfigure/2.3.1.RELEASE/6d679d6ba26235a0e81ca1d58f9c1024d9427411/spring-boot-autoconfigure-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot/2.3.1.RELEASE/ce8d8b6838ecceb68962b975b18682f4237ccf71/spring-boot-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-starter-logging/2.3.1.RELEASE/3f242a91ffddf7485fde1367e1354c7e13024c8/spring-boot-starter-logging-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/jakarta.annotation/jakarta.annotation-api/1.3.5/59eb84ee0d616332ff44aba065f3888cf002cd2d/jakarta.annotation-api-1.3.5.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-core/5.2.7.RELEASE/56e14a3a5e2813534b5db2da1502cd58ab5bc61d/spring-core-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.yaml/snakeyaml/1.26/a78a8747147d2c5807683e76ec2b633e95c14fe9/snakeyaml-1.26.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-jdbc/5.2.7.RELEASE/847d31c90479a34e4e1fe7eeeb47ac89adce3438/spring-jdbc-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/com.zaxxer/HikariCP/3.4.5/aa1a2c00aae8e4ba8308e19940711bb9525b103d/HikariCP-3.4.5.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-context/5.2.7.RELEASE/7fd9c4ea311a5d9ab92770be7fc93cc53db334f9/spring-context-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-aop/5.2.7.RELEASE/9cf69f8e888091684c05f0a287bb638502e90725/spring-aop-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-orm/5.2.7.RELEASE/32a76d825d8782ff278abe124ded9620444b4a74/spring-orm-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework.data/spring-data-commons/2.3.1.RELEASE/5ae66c24c223315d5b31a45590d293e9145c18e9/spring-data-commons-2.3.1.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-tx/5.2.7.RELEASE/66faebf0da41c67b67d082efc98e92c40e83e6b/spring-tx-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-api/1.7.30/b5a4b6d16ab13e34a88fae84c35cd5d68cac922c/slf4j-api-1.7.30.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/antlr/antlr/2.7.7/83cd2cd674a217ade95a4bb83a8a14f351f48bd0/antlr-2.7.7.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/net.bytebuddy/byte-buddy/1.10.11/16ac7e0d4afef10ac30db377e8151aff66a90e1c/byte-buddy-1.10.11.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/com.fasterxml/classmate/1.5.1/3fe0bed568c62df5e89f4f174c101eab25345b6c/classmate-1.5.1.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.glassfish.jaxb/jaxb-runtime/2.3.3/c46b68a6e3a2d84ba4eb14c6a8a1a9a7be4048bc/jaxb-runtime-2.3.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.hibernate.common/hibernate-commons-annotations/5.1.0.Final/700aeedc4a2089816621948f0379e17cbd17d5db/hibernate-commons-annotations-5.1.0.Final.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.jboss.logging/jboss-logging/3.4.1.Final/40fd4d696c55793e996d1ff3c475833f836c2498/jboss-logging-3.4.1.Final.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.javassist/javassist/3.24.0-GA/d7466fc2e3af7c023e95c510f06448ad29b225b3/javassist-3.24.0-GA.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.jboss/jandex/2.1.3.Final/cd56603e39eb1421560b71daa584348ecfd9e0b8/jandex-2.1.3.Final.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.dom4j/dom4j/2.1.3/a75914155a9f5808963170ec20653668a2ffd2fd/dom4j-2.1.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.aspectj/aspectjweaver/1.9.5/1740dc9140103b796d1722668805fd4cf852780c/aspectjweaver-1.9.5.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-to-slf4j/2.13.3/966f6fd1af4959d6b12bfa880121d4a2b164f857/log4j-to-slf4j-2.13.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/ch.qos.logback/logback-classic/1.2.3/7c4f3c474fb2c041d8028740440937705ebb473a/logback-classic-1.2.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.slf4j/jul-to-slf4j/1.7.30/d58bebff8cbf70ff52b59208586095f467656c30/jul-to-slf4j-1.7.30.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-jcl/5.2.7.RELEASE/72282e1f89c58284632220437b5a1e8066c53d7d/spring-jcl-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-beans/5.2.7.RELEASE/5465ab17688ed62254fdef411cf883fd5c3b77a/spring-beans-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.springframework/spring-expression/5.2.7.RELEASE/c98d7b10f959f9bedfbbbd4d723cf7a1f17a1f71/spring-expression-5.2.7.RELEASE.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.glassfish.jaxb/txw2/2.3.3/12f70b0ea4fc1ad45315e842f63f7c9a46f46530/txw2-2.3.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/jakarta.xml.bind/jakarta.xml.bind-api/2.3.3/48e3b9cfc10752fba3521d6511f4165bea951801/jakarta.xml.bind-api-2.3.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/com.sun.istack/istack-commons-runtime/3.0.11/4293b5f4e4e89d598f62bb2ba73b32132e7c3a27/istack-commons-runtime-3.0.11.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-api/2.13.3/ec1508160b93d274b1add34419b897bae84c6ca9/log4j-api-2.13.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/ch.qos.logback/logback-core/1.2.3/864344400c3d4d92dfeb0a305dc87d953677c03c/logback-core-1.2.3.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/org.postgresql/postgresql/42.2.14/45fa6eef266aa80024ef2ab3688d9faa38c642e5/postgresql-42.2.14.jar:/Users/ionutursuleanu/.gradle/caches/modules-2/files-2.1/com.sun.activation/jakarta.activation/1.2.2/74548703f9851017ce2f556066659438019e7eb5/jakarta.activation-1.2.2.jar com.example.demo.DemoApplication
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
2020-07-16 22:35:50.262 INFO  org.glowroot - Glowroot version: 0.13.7-SNAPSHOT, built 2020-07-15 17:28:01 +0000
2020-07-16 22:35:50.266 INFO  org.glowroot - Java version: 14.0.1 (AdoptOpenJDK / Mac OS X)
2020-07-16 22:35:50.268 INFO  org.glowroot - Java args: -javaagent:/Users/ionutursuleanu/glowroot/glowroot.jar -XX:TieredStopAtLevel=1 -Xverify:none -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=49899:/Applications/IntelliJ IDEA.app/Contents/bin
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
2020-07-16 22:35:52.868 INFO  org.glowroot - UI listening on 127.0.0.1:4000 (to access the UI from remote machines, change the bind address to 0.0.0.0, either in the Glowroot UI under Configuration > Web or directly in the admin.json file, and then restart JVM to take effect)

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.3.1.RELEASE)

2020-07-16 22:35:53.319  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Starting DemoApplication on 192-168-0-103.rdsnet.ro with PID 1059 (/Users/ionutursuleanu/IdeaProjects/demo/build/classes/java/main started by ionutursuleanu in /Users/ionutursuleanu/IdeaProjects/demo)
2020-07-16 22:35:53.324  INFO 1059 --- [           main] com.example.demo.DemoApplication         : No active profile set, falling back to default profiles: default
2020-07-16 22:35:54.073  INFO 1059 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFERRED mode.
2020-07-16 22:35:54.143  INFO 1059 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 61ms. Found 1 JPA repository interfaces.
2020-07-16 22:35:54.697  INFO 1059 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-07-16 22:35:54.812  INFO 1059 --- [         task-1] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2020-07-16 22:35:54.861  INFO 1059 --- [           main] DeferredRepositoryInitializationListener : Triggering deferred initialization of Spring Data repositories…
2020-07-16 22:35:54.902  INFO 1059 --- [         task-1] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.4.17.Final
2020-07-16 22:35:55.065  INFO 1059 --- [         task-1] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.0.Final}
2020-07-16 22:35:55.192  INFO 1059 --- [         task-1] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2020-07-16 22:35:55.335  INFO 1059 --- [         task-1] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2020-07-16 22:35:55.385  INFO 1059 --- [         task-1] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.PostgreSQL10Dialect
2020-07-16 22:35:56.103  INFO 1059 --- [         task-1] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2020-07-16 22:35:56.110  INFO 1059 --- [         task-1] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2020-07-16 22:35:56.457  INFO 1059 --- [           main] DeferredRepositoryInitializationListener : Spring Data repositories initialized!
2020-07-16 22:35:56.465  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Started DemoApplication in 3.684 seconds (JVM running for 6.713)
2020-07-16 22:35:56.544  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customers found with findAll():
2020-07-16 22:35:56.544  INFO 1059 --- [           main] com.example.demo.DemoApplication         : -------------------------------
2020-07-16 22:35:56.651  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=1, firstName='Jack', lastName='Bauer']
2020-07-16 22:35:56.652  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=2, firstName='Chloe', lastName='O'Brian']
2020-07-16 22:35:56.652  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=3, firstName='Kim', lastName='Bauer']
2020-07-16 22:35:56.652  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=4, firstName='David', lastName='Palmer']
2020-07-16 22:35:56.652  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=5, firstName='Michelle', lastName='Dessler']
2020-07-16 22:35:56.652  INFO 1059 --- [           main] com.example.demo.DemoApplication         : 
2020-07-16 22:35:56.663  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer found with findById(1L):
2020-07-16 22:35:56.664  INFO 1059 --- [           main] com.example.demo.DemoApplication         : --------------------------------
2020-07-16 22:35:56.665  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=1, firstName='Jack', lastName='Bauer']
2020-07-16 22:35:56.665  INFO 1059 --- [           main] com.example.demo.DemoApplication         : 
2020-07-16 22:35:56.665  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer found with findByLastName('Bauer'):
2020-07-16 22:35:56.665  INFO 1059 --- [           main] com.example.demo.DemoApplication         : --------------------------------------------
2020-07-16 22:35:56.746  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=1, firstName='Jack', lastName='Bauer']
2020-07-16 22:35:56.749  INFO 1059 --- [           main] com.example.demo.DemoApplication         : Customer[id=3, firstName='Kim', lastName='Bauer']
2020-07-16 22:35:56.754  INFO 1059 --- [           main] com.example.demo.DemoApplication         : 

```

**Comment by nezda** [comment]
Aha! If I add `-Xverify:none` like you have there my app works too but that seems weird / potentially buggy? (Although I do see the warning `OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.`)

**Comment by ionutursuleanu** [comment]
@trask could you help us with this one? Following your idea that asm dependency needs to be updated I reported this error to asm project(after trying asm 8.01 & 9 beta) and this was the answer: [https://gitlab.ow2.org/asm/asm/-/issues/317914#note_55924](https://gitlab.ow2.org/asm/asm/-/issues/317914#note_55924) What do you think?

**Comment by trask** [comment]
Hey @ionutursuleanu, can you try to create a (failing) test for this, similar to the Java 8 weaving tests here:

https://github.com/glowroot/glowroot/tree/master/agent/core/src/test/java8/org/glowroot/agent/weaving

but for Java 14?

That would make it much easier for me to debug and fix.

**Comment by ionutursuleanu** [comment]
Hi @trask ,

I created #741 with the failing test

**Comment by matizerix** [comment]
Hi @trask and @ionutursuleanu,
are there any news with this issue?

I'm facing the same problem with glowroot and jdk14 with a spring application on tomcat 9.

Worked fine until I switched from jdk8 to 14.

**Comment by ionutursuleanu** [comment]
Hi @matizerix ,

The issue is related to bytecode manipulation...that's the only thing I know. My knowledge in this area is basically zero, so I can't help you.

**Comment by matizerix** [comment]
@ionutursuleanu thanks for your response. In my environment your test https://github.com/glowroot/glowroot/pull/741 doesn't fail. Does it actually fail in your environment with a VerifyError?
Thanks!

**Comment by ionutursuleanu** [comment]
> In my environment your test #741 doesn't fail. Does it actually fail in your environment with a VerifyError?

Yes, it fails...but only if it's running with java 14/15. Try to build the project with java 8/11 but run the test with 14/15. 

You could also checkout https://github.com/ionutursuleanu/glowroot/tree/jdk7build (building with java 14 requires 1.7 source/target level in maven) & https://github.com/ionutursuleanu/glowroot/tree/java14-verifyerror

**Comment by nezda** [comment]
@matizerix I think you could add `-Xverify:none` JVM flag

https://github.com/glowroot/glowroot/issues/738#issuecomment-659700974

**Comment by mschaaf** [comment]
> > In my environment your test #741 doesn't fail. Does it actually fail in your environment with a VerifyError?
> 
> Yes, it fails...but only if it's running with java 14/15. Try to build the project with java 8/11 but run the test with 14/15.
> 
> You could also checkout https://github.com/ionutursuleanu/glowroot/tree/jdk7build (building with java 14 requires 1.7 source/target level in maven) & https://github.com/ionutursuleanu/glowroot/tree/java14-verifyerror

@ionutursuleanu does that mean if a release is compiled with JDK 14/15 then it would not fail? If yes would it then be a solution to compile a release with JDK 14/15 for usage with JDK 14/15?

**Comment by jenni** [comment]
+1

**Comment by matizerix** [comment]
> @matizerix I think you could add `-Xverify:none` JVM flag
> 
> [#738 (comment)](https://github.com/glowroot/glowroot/issues/738#issuecomment-659700974)

@nezda  Thanks for the suggestion, that would work but I prefere for now, to not completely disable bytecode verification.

**Comment by matizerix** [comment]
@mschaaf yes that does work. But i had to do a couple of changes to the source code and disable some tests. If you are interested  I may push the changes to my personal repo.

**Comment by mschaaf** [comment]
@matizerix Yes, please share your changes with me. I would like to try it out.

**Comment by matizerix** [comment]
> 
> 
> @mschaaf yes that does work. But i had to do a couple of changes to the source code and disable some tests. If you are interested I may push the changes to my personal repo.
@mschaaf 
Arrr... just found out that my colleagues added verifiy:none argument and my changes do actually nuts :( Sorry for the false hope.

**Comment by stefpiatek** [comment]
May want to check to see if the 0.14.0-beta.2 release fixes this, I had a similar error and its now working using it

**Comment by bcourtine** [comment]
> May want to check to see if the 0.14.0-beta.2 release fixes this, I had a similar error and its now working using it

I agree. I had a similar problem with jdk 14+ until 0.14.0-beta.1, and it disappeared in 0.14.0-beta.2.
I think the fix #859 solved this VerifyError issue, and I can now use Glowroot without error on JDK 17.

**Comment by eostermueller** [comment]
0.14.0-beta.3 fixed it for me, appreciate the work on this....this entire project.
